lucene/core/src/java/org/apache/lucene/index/DefaultIndexingChain.java

```diff
@@ -527,45 +589,61 @@ private void indexPoint(PerField fp, IndexableField field) throws IOException {
     fp.pointValuesWriter.addPackedValue(docState.docID, field.binaryValue());
   }
 
-  private void validateIndexSortDVType(Sort indexSort, String fieldName, DocValuesType dvType) {
+  private void validateIndexSortDVType(Sort indexSort, String fieldToValidate, DocValuesType dvType) throws IOException {
     for (SortField sortField : indexSort.getSort()) {
-      if (sortField.getField().equals(fieldName)) {
-        switch (dvType) {
-          case NUMERIC:
-            if (sortField.getType().equals(SortField.Type.INT) == false &&
-                  sortField.getType().equals(SortField.Type.LONG) == false &&
-                  sortField.getType().equals(SortField.Type.FLOAT) == false &&
-                  sortField.getType().equals(SortField.Type.DOUBLE) == false) {
-              throw new IllegalArgumentException("invalid doc value type:" + dvType + " for sortField:" + sortField);
-            }
-            break;
+      IndexSorter sorter = sortField.getIndexSorter();
+      assert sorter != null;
```

Comment:
should it be an actual exception instead of an assertion? Or are we checking it up-front already?

Reply:
It's checked in `IndexWriterConfig.setSort()` but I'll add an exception here as well; you could conceivably get here via a funky Codec that instantiates a Sort with no IndexSorter defined.